libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/widgets/WPBottomSheetDialogFragment.java

```diff
@@ -0,0 +1,58 @@
+package org.wordpress.android.login.widgets;
+
+import android.app.Dialog;
+import android.content.res.Resources;
+import android.os.Bundle;
+import android.view.Gravity;
+import android.widget.FrameLayout;
+
+import androidx.annotation.NonNull;
+import androidx.coordinatorlayout.widget.CoordinatorLayout;
+
+import com.google.android.material.bottomsheet.BottomSheetDialog;
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
+
+import org.wordpress.android.login.R;
+
+public class WPBottomSheetDialogFragment extends BottomSheetDialogFragment {
+    @Override
+    public int getTheme() {
+        return R.style.LoginTheme_BottomSheetDialogStyle;
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        return new BottomSheetDialog(requireContext(), getTheme());
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        Dialog dialog = getDialog();
+        if (dialog != null) {
+            restrictMaxWidthForDialog(dialog);
+        }
+    }
+
+    private void restrictMaxWidthForDialog(@NonNull Dialog dialog) {
+        Resources resources = dialog.getContext().getResources();
+        int dp = (int) resources.getDimension(R.dimen.bottom_sheet_dialog_width);
+        // Limit width of bottom sheet on wide screens; non-zero width defined only for large qualifier.
+        if (dp > 0) {
+            FrameLayout bottomSheetLayout =
+                    dialog.findViewById(com.google.android.material.R.id.design_bottom_sheet);
```

Comment:
>If that resource name is changed or removed, it can cause a NullPointerException crash from the bottomSheetLayout.getLayoutParams() call in the line below

We include the sources of the library in our APKs, so I believe it won't crash in runtime. 
We'll find out when we decide to update the version of the library that the resources is missing because the project won't build, right?
I agree that it's still considered an anti-pattern, but this whole customization of a component which isn't supposed to be customized is an anti-pattern. So I personally wouldn't worry about it too much unless we have an easy solution for it.
Wdyt?

Reply:
If the resource is removed completely, it should cause a build error.  If the resource is moved to another view or layout, it would be a bug or crash.  The point is that we don't have control over the library resources.  Adding a `null`-check is just a safety measure.

Reply:
👍 yeah, adding null check is definitely a good call! Completely agree ;). Thanks for the elaboration:)